components/ILIAS/Registration/src/DualOptIn/Entity/PendingRegistration.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\DualOptIn\Entity;
+
+use DateTimeImmutable;
+use ILIAS\Data\ObjectId;
+use ILIAS\Data\UUID\RamseyUuidWrapper;
+
+final class PendingRegistration
+{
+    private RamseyUuidWrapper $id;
+    private ObjectId $usr_id;
+    private RegistrationHash $hash;
+    private DateTimeImmutable $created_at;
+
+    public function __construct(
+        RamseyUuidWrapper $id,
```

Comment:
We should not depend on the `RamseyUuidWrapper`. Could we introduce a new class `PedingRegistrationI`, which encapsulates the `UUID` (the simple string, not the `RamseyUuidWrapper`)?